### PR TITLE
Formatted project

### PR DESCRIPTION
**Pull request by BenjaminIsMyName**
The things I did in this one simple commit was:
1. Adding config files - `.editorconfig` and `prettier.config.js`
2. Formatted all the files in the project with the new settings
3. That's it

These config files are important, to prevent unnecessary changes when people create PRs with a simple fix but format the file when saving, causing extra lines to change and extra work when reviewing the PRs. 

For example, [in this file on a PR request](https://github.com/bryantobing12/layout-reproduce-i18next/pull/1/files#diff-fce332fdab5dc79594ac42cc8cd6114832222da4aac765842742fe1e0ddd0a55) you can clearly see changes that were made only due to formatter settings. 

This is my first PR on a big project, be nice :-)

### CODE REVIEW

**Comment by belgattitude** [comment]
I'm for it. Also realized that it wasn't so easy to work in the repo, so I'm quite positive about the changes

On what I see prettier and eslint are in disagreement, so I guess there's still some tuning to do. Can't really explain today, but there's some material that can help here: https://github.com/belgattitude/shared-dx/tree/main/packages/eslint-config-bases#prettier-integration.

Personnally I wouln't add editorconfig, as most IDE today are able to read the prettier config.

**Comment by belgattitude** [comment]
Could you commit the prettier config as well ?

**Comment by BenjaminIsMyName** [comment]
> Could you add prettier to devDeps and commit the prettier config as well ? 

Sure, will work on it & dig into why ESLint causing troubles in the upcoming week.

**Comment by BenjaminIsMyName** [comment]
@belgattitude looks good now? 
Resolved all the conflicts between `prettier` and `ESLint` & installed `prettier` as dev-dep.
I didn't remove `.editorconfig` because I think it's (in the worst case scenario) harmless. And it's already there, so why not...

**Review by belgattitude — APPROVED** [review]
Looks good to me. 

Personal note: I'm preferring the "semi: true" (https://prettier.io/docs/en/options.html#semicolons) but I know nextjs itself prefers without. So it makes sense

**Review by belgattitude — COMMENTED** [review]
just to check a weird formatting

File: src/appWithTranslation.client.test.tsx [hunk 1]
```diff
@@ -31,49 +31,51 @@ jest.mock('react-i18next', () => ({
...
+  render(<DummyApp {...props} />)
 
 describe('appWithTranslation', () => {
   beforeEach(() => {
-    (fs.existsSync as jest.Mock).mockReturnValue(true);
-    (fs.readdirSync as jest.Mock).mockReturnValue([]);
-    (I18nextProvider as jest.Mock).mockImplementation(DummyI18nextProvider)
-    const actualCreateClient = jest.requireActual('./createClient');
-    (createClient as jest.Mock).mockImplementation(actualCreateClient)
+    ;(fs.existsSync as jest.Mock).mockReturnValue(true)
```
[the remark this answers is not in the document]
Comment (BenjaminIsMyName) [reply src/appWithTranslation.client.test.tsx:70]: See here: https://github.com/prettier/prettier/issues/1907#issuecomment-305910670

**Comment by belgattitude** [comment]
@BenjaminIsMyName Just to let you know, I'm investigating why our ci checks haven't been able to run. One i'm sure I'll merge

File: src/appWithTranslation.client.test.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (belgattitude) [reply src/appWithTranslation.client.test.tsx:70]: Thx, got it. One reason more to prefer semi: true imho :smile: 

Let's see if we change that in the future

**Pull request merged**